oak_functions_launcher/src/server.rs

```diff
@@ -15,59 +15,46 @@
 //
 
 use crate::{channel::ConnectorHandle, schema};
-use futures::Future;
-use oak_remote_attestation_sessions::{SessionId, SESSION_ID_LENGTH};
-use std::net::SocketAddr;
-use tonic::{transport::Server, Request, Response};
-
-pub mod proto {
-    tonic::include_proto!("oak.session.unary.v1");
-}
-
-use proto::{
-    unary_session_server::{UnarySession, UnarySessionServer},
-    PublicKeyInfo, UnaryRequest, UnaryResponse,
+use futures::{Future, Stream, StreamExt};
+use oak_remote_attestation_noninteractive::proto::{
+    request_wrapper, response_wrapper, streaming_session_server::StreamingSessionServer,
+    InvokeResponse, RequestWrapper, ResponseWrapper,
 };
-
-fn encode_request(unary_request: UnaryRequest) -> Result<schema::UserRequest, oak_idl::Status> {
-    let mut session_id: SessionId = [0; SESSION_ID_LENGTH];
-    if unary_request.session_id.len() != SESSION_ID_LENGTH {
-        return Err(oak_idl::Status::new_with_message(
-            oak_idl::StatusCode::Internal,
-            format!(
-                "session_id must be {} bytes in length, found a length of {} bytes instead",
-                SESSION_ID_LENGTH,
-                unary_request.session_id.len()
-            ),
-        ));
-    }
-    session_id.copy_from_slice(&unary_request.session_id);
-
-    // Return the underlying owned buffer
-    Ok(schema::UserRequest {
-        session_id: Some(schema::SessionId {
-            value: session_id.to_vec(),
-        }),
-        body: unary_request.body,
-    })
-}
+use std::{net::SocketAddr, pin::Pin};
+use tonic::{transport::Server, Request, Response, Status, Streaming};
 
 pub struct SessionProxy {
     connector_handle: ConnectorHandle,
-    signing_public_key: Vec<u8>,
-    attestation: Vec<u8>,
+    // TODO(#3442): Return cached key and attestation when clients ask for it.
+    _signing_public_key: Vec<u8>,
+    _attestation: Vec<u8>,
 }
 
 #[tonic::async_trait]
-impl UnarySession for SessionProxy {
-    async fn message(
+impl oak_remote_attestation_noninteractive::proto::streaming_session_server::StreamingSession
```

Comment:
Once we add both interactive and non-interactive attestation, will they be implemented as different RPC services?

Reply:
yes, since the interactive one will never be exposed externally, right?